debezium-core/src/main/java/io/debezium/pipeline/spi/OffsetContext.java

```diff
@@ -57,4 +60,11 @@
      * Signals that a snapshot has been completed, which should reflect in an updated offset state.
      */
     void postSnapshotCompletion();
+
+    /**
+     * Records the name of the table and the timestamp of the last event
+     * @param tableId
+     * @param timestamp
+     */
+    void event(TableId tableId, Instant timestamp);
```

Comment:
I only see this being called once, in the `RelationalSnapshotChangeEventSource`; why not during streaming?

Reply:
Beacuse this is ccommon onfly for snapshot. In streaming every connector has its own streaming context metadata.

Reply:
Ok, can we find a more descriptive name then? And can you check with the SQL Server connector, I can't find where source time and table id are advanced during streaming for that one now.

Reply:
@gunnarmorling ok, I stand corrected the `evetn` method is used in`io.debezium.connector.sqlserver.SqlServerStreamingChangeEventSource.execute(ChangeEventSourceContext)`

Just in PostgresConnectorOffset there is a single method call updating all metadata at once. In case of SqlSetrver there is call of two methods - `event` and `setChangePosition`. So I propose to keep it as is and as a follow-up task to introduce new or extend the `event` method with a parameterized type parameter that would accept `Coordinates` class that will be specific per-connector and contain table, timestampt and other fields.

Reply:
Done: https://issues.jboss.org/browse/DBZ-1435